pyiode/iode_database/abstract_database.pyx

```diff
@@ -100,32 +100,24 @@ cdef class _AbstractDatabase:
         >>> comments.load(f"{SAMPLE_DATA_DIR}/fun.cmt")
         >>> comments.is_subset()
         False
-        >>> # by default a 'shallow copy' subset is returned
-        >>> cmt_subset = comments.subset("A*")
+        >>> cmt_subset = comments.copy("A*")
         >>> cmt_subset.is_subset()
         True
         >>> cmt_subset.is_copy_subset()
-        False
-        >>> # force to return a 'deep copy' subset
-        >>> cmt_subset = comments.subset("A*", copy=True)
-        >>> cmt_subset.is_copy_subset()
         True
         """
         if self.abstract_db_ptr.is_global_database():
             return False
         return self.abstract_db_ptr.is_local_database()
 
-    def subset(pattern: str, copy: bool = False) -> Self:
-        """
-        Create a subset of a global IODE database.
-
-        If 'copy' is True, a *deep copy* subset is returned. 
-        This means that any change made on an object of the subset will **NOT** 
-        modify the corresponding object in the global IODE database.
+    def _subset(self, pattern: str, copy: bool) -> Self:
+        raise NotImplementedError()
 
-        If 'copy' is False (default), a *shallow copy* subset is returned. 
-        This means that any change made on an object of the subset will also 
-        modify the corresponding object in the global IODE database.
+    def copy(self, pattern: str) -> Self:
```

Comment:
I don't think users will frequently call `copy()`. Users are got used to work with "workspaces", meaning a global database for each IODE object type. As you suggest, it would be more logical to remove pattern as argument and encourage users to do `database[pattern].copy()`